source/plsql/cache/nocopy-directive.rst

```diff
@@ -24,7 +24,7 @@ The ``NOCOPY`` hint can thus help you reduce unnecessary CPU cycles and memory c
 There are several cases where you can be sure the compiler ignores ``NOCOPY``:
  
 * The actual parameter must be *implicitly* converted to the data type of the formal parameter.
-* The actual parameter is the element of a collection.
+* The actual parameter is the element of a collection (unless it is also a collection -- see dbms_sql.column_value procedure).
```

Comment:
How can a parameter be both an element of a collection and a collection at the same time? 

I don't think this sentence makes any sense.

See also the [official doc](https://docs.oracle.com/database/121/LNPLS/formal_parameter.htm#GUID-5BA8E033-96B9-439A-A4FC-4844FEC14AD8__CJACJCGJ).

Reply:
arrays of arrays, that's how.  I concede it is not clear and my feelings will not be hurt if you reject the pull request and update it another way.

This is the definition of the column_value procedure:

DBMS_SQL.COLUMN_VALUE(
 c IN INTEGER,
 position IN INTEGER,
 <param_name> IN OUT NOCOPY <table_type>);

The parameter passed into this procedure is a collection type.  If I have another collection whose elements are collections, then I can pass an element of that collection into this procedure and it is provided as a pointer/references and the NOCOPY hint works just fine. I know because I tried it.

DECLARE
    v_clob_tab DBMS_SQL.clob_table := DBMS_SQL.clob_table();
    TYPE t_arr_clob_tab IS TABLE OF DBMS_SQL.clob_table INDEX BY BINARY_INTEGER:
    v_arr_clob_tab t_arr_clob_tab;
BEGIN
v_arr_clob_tab(3) := v_clob_tab; -- assign empty array
DBMS_SQL.column_value(ctx, 3, v_arr_clob_tab(3));

Reply:
The word "it" was too vague. I replaced with "the element" which I hope makes it more clear.

Honestly, you can just close the pull request and ignore it. I was searching the web to try to figure out if nocopy would work in the scenario I was trying to use it, and ran across your very good writeup. It almost made me turn away from my path, but fortunately I remembered reading something about "collections were special with regard to reference vs copy" and tried it anyway.  Your article is near the top of a google search. Thought I would try to help the next poor slob.